src/vtquery.cpp

```diff
@@ -476,7 +476,7 @@ NAN_METHOD(vtquery) {
         if (!z_val->IsNumber()) {
             return utils::CallbackError("'z' value in 'tiles' array item is not a number", callback);
         }
-        std::int64_t z = z_val->IntegerValue();
+        int z = z_val->Int32Value();
```

Comment:
What happens now if someone passes the equivalent of `std::numeric_limits<std::int64_t>::max()` (aka `9223372036854775807`) in from JS land? What will `Int32Value()` return? Do we have a test for this? /cc @mapsam

Reply:
Good question - @artemp have you covered this in vtcomposite?

Reply:
@mapsam @springmeyer - we need to change `isNumber()` to `isInt32()` to avoid undefined behaviour